packages/utils/src/schema/toPathSchema.ts

```diff
@@ -37,8 +37,9 @@ export default function toPathSchema<T = any>(
     $name: name.replace(/^\./, ""),
   } as PathSchema;
 
-  if (ADDITIONAL_PROPERTIES_KEY in schema) {
-    set(pathSchema, "__rjsf_additionalProperties", true);
+  if (ADDITIONAL_PROPERTIES_KEY in schema && 
+    schema[ADDITIONAL_PROPERTIES_KEY] === true) {
+    set(pathSchema, '__rjsf_additionalProperties', true);
```

Comment:
These want to be double quotes and will be updated automatically by the formatter